src/uct/cuda/base/cuda_md.c

```diff
@@ -85,7 +77,12 @@ uct_cuda_base_get_base_addr_alloc_length(uct_cuda_copy_md_t *md,
         if (cu_err != CUDA_SUCCESS) {
             cuGetErrorString(cu_err, &cu_err_str);
             ucs_error("cuDeviceGetName error: %s", cu_err_str);
-            return UCS_ERR_INVALID_ADDR;
+            goto err;
+        }
+
+        if (!strncmp(dev_name[cuda_device], "T4", 2)) {
```

Comment:
no need to save a global array of device names. total_bytes is enough

Reply:
@yosefe total_bytes for t4 would be ~4GB. Any allocation less than 400MB would be a valid candidate for registration but t4 BAR1 space limit is 256 MB. For this reason, I think we shouldn't allow some devices to use this feature unless forced through env var. WDYT?

Reply:
@yosefe Addressed this now. Lmk if the changes look good.